packages/react/src/hooks/utils/useInvalidateOnBlock.ts

```diff
@@ -9,7 +11,7 @@ export function useInvalidateOnBlock({
   enabled?: boolean
   queryKey: QueryKey
 }) {
-  const queryClient = useQueryClient()
+  const queryClient = useQueryClient({ context: queryClientContext })
```

Comment:
Can we have our own `useQueryClient` in `utils/query` that uses `context` so we don't have to manually pass it on each usage?